### PR TITLE
feat: add PIN Support

### PR DESCRIPTION
**Pull request by pavan5555**
- Added PIN support 

- Displays PIN along with the Link generation.
- Displays the PIN Valid for 30 minutes msg

image

### CODE REVIEW

File: client/src/components/LinkDisplay/index.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React, { useRef, useState, useContext } from 'react';
 import { FiLink, FiCopy, FiExternalLink } from 'react-icons/fi';
 import styles from './Style.module.css';
 import { ThemeContext } from '../../ThemeContext.js';
+import PinDisplay from './../PinDisplay/index.js';
```
Comment (muke1908) [inline client/src/components/LinkDisplay/index.js:5]:
```suggestion
import PinDisplay from './../PinDisplay';
```

**Review by muke1908 — CHANGES_REQUESTED** [review]

**Comment by muke1908** [comment]
@pavan5555 thanks for the PR, wondering if you add the same to the chat page as well? 
Screen Shot 2020-10-07 at 9 06 56 AM

**Comment by pavan5555** [comment]
Yes!, will you create a new issue ? or I use the same issue ?, so sorry about these questions.

**Comment by muke1908** [comment]
> Yes!, will you create a new issue ? or I use the same issue ?, so sorry about these questions.

Let's do it in the scope of this issue.

**Comment by pavan5555** [comment]
> > Yes!, will you create a new issue ? or I use the same issue ?, so sorry about these questions.
> 
> Let's do it in the scope of this issue.

Sure!

**Comment by pavan5555** [comment]
Hello @muke1908  I have pushed the changes

- Added PIN Display in chat page 
- Worked on the CSS issue in the chat page in light mode.

image

File: client/src/components/LinkDisplay/index.js [hunk 2]
```diff
@@ -45,11 +46,22 @@ const LinkDisplay = ({ content }) => {
...
+        <PinDisplay content={content.pin} />
+        <span className={styles.pinValidMsg}>PIN is valid for 30 minutes</span>
+      </div>
       <div
         className={`${styles.openLink}
       ${darkMode ? styles.darkOpenLink : styles.lightOpenLink}`}
       >
-        <a href={content} target="_blank" rel="noopener noreferrer">
+        <a
+          href={`${content.absoluteLink}?pin=${content.pin}`}
```
Comment (muke1908) [inline client/src/components/LinkDisplay/index.js:61]: Adding PIN in url parameter - it's a security vulnerability. 
Please add a **To Do** comment that "this needs to be changed - make an POST endpoint to fetch the PIN of an url"

**Review by muke1908 — CHANGES_REQUESTED** [review]
Please resolve the conflicts.

**Pull request merged**